sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java

```diff
@@ -166,12 +170,17 @@ Mono<StoreResponse> writePrivateAsync(
 
             Mono<List<AddressInformation>> replicaAddressesObs = this.addressSelector.resolveAddressesAsync(request, forceRefresh);
             AtomicReference<Uri> primaryURI = new AtomicReference<>();
-            AtomicReference<List<String>> replicaStatusList = new AtomicReference<>();
+            Map<String, List<String>> replicaStatusList = new ConcurrentHashMap<>();
+            Map<Uri, String> replicaStatuses = new ConcurrentHashMap<>();
 
             return replicaAddressesObs.flatMap(replicaAddresses -> {
                 try {
                     List<URI> contactedReplicas = new ArrayList<>();
-                    replicaAddresses.forEach(replicaAddress -> contactedReplicas.add(replicaAddress.getPhysicalUri().getURI()));
+                    replicaAddresses.forEach(replicaAddress -> {
+                        Uri uri = replicaAddress.getPhysicalUri();
+                        contactedReplicas.add(uri.getURI());
+                        replicaStatuses.put(uri, uri.getHealthStatusDiagnosticString());
```

Comment:
why not just make this conditional on uri not being primaryUri - and then avoid the remove again below?

Reply:
Changed to make it conditional on primaryUri